dask_jobqueue/cluster_manager.py

```diff
@@ -35,7 +34,8 @@ def _scale(self, n):
                 to_close = self.scheduler.workers_to_close(
                     n=len(self.scheduler.workers) - n)
                 logger.debug("Closing workers: %s", to_close)
-                self.scheduler.retire_workers(workers=to_close)
+                # Should  be an RPC call here
+                yield self.scheduler.retire_workers(workers=to_close)
```

Comment:
As a warning, now that you've added a `yield` in this coroutine it's entirely possible for another coroutine to start running while this one waits for a response.  It is entirely possible that two `_scale` coroutines will be active at the same time.

You still can't use a `threading.Lock` to fix this (threading locks will lock the entire event loop).  You can use a Tornado lock, or a few other methods.  Short term I wouldn't worry about it though.

Reply:
Thanks for the precision. I don't think this is an issue yet.